index.d.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * Notes about these type definitions:
+ *
+ * - Callbacks returning multiple completion values using multiple arguments are not supported by these types.
+ *   Prefer to use Node's style by grouping your values in a single object or array.
+ *   Support for this kind of callback is blocked by Microsoft/TypeScript#5453
+ *
+ * - For ease of use, `asyncDone` lets you pass callback functions with a result type `T` instead of `T | undefined`.
+ *   This matches Node's types but can lead to unsound code being typechecked.
+ *
+ *   The following code typechecks but fails at runtime:
+ *   ```typescript
+ *   async function getString(): Promise<string> {
+ *     return "Hello, World!";
+ *   }
+ *
+ *   async function evilGetString(): Promise<string> {
+ *     throw new Error("Hello, World!");
+ *   }
+ *
+ *   function cb(err: Error | null, result: string): void {
+ *     // This is unsound because `result` is `undefined` when `err` is not `null`.
+ *     console.log(result.toLowerCase());
+ *   }
+ *
+ *   asyncDone(getString, cb); // Prints `hello, world!`
+ *   asyncDone(evilGetString, cb); // Runtime error: `TypeError: Cannot read property 'toLowerCase' of undefined`
+ *   ```
+ *
+ *   Enforcing stricter callbacks would require developers to use `result?: string` and assert the existence
+ *   of the result either by checking it directly or using the `!` assertion operator after testing for errors.
+ *   ```typescript
+ *   function stricterCb1(err: Error | null, result?: string): void {
+ *     if (err !== null) {
+ *       console.error(err);
+ *       return;
+ *     }
+ *     console.log(result!.toLowerCase());
+ *   }
+ *
+ *   function stricterCb2(err: Error | null, result?: string): void {
+ *     if (result === undefined) {
+ *       console.error("Undefined result. Error:);
+ *       console.error(err);
+ *       return;
+ *     }
+ *     console.log(result.toLowerCase());
+ *   }
+ *   ```
+ */
+import { ChildProcess } from "child_process";
+import { EventEmitter } from "events";
+import { Readable as ReadableStream } from "stream";
```

Comment:
This can be Readable, Writeable or Transform/Duplex because we exhaust the stream.

Reply:
I think it might be able to reference the base "Stream" instead?  Not sure if it contains the indicators we use to check for stream.

Reply:
I originally had `Stream` but wasn't sure if Writable streams were supported. Thanks for confirming that any Node stream is fine. I'll update it.